dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavResult.java

```diff
@@ -0,0 +1,374 @@
+
+public class NavResult implements Iterable<NavResult>, Permissionable, Serializable {
+
+
+    private static final long serialVersionUID = 1L;
+    private long languageId;
+    public long getLanguageId() {
+        return languageId;
+    }
+
+    private String title;
+    private String href;
+    private int order;
+    private String codeLink;
+    private String parent;
+    private String type;
+    private String permissionId;
+    private String target;
+    private boolean showOnMenu;
+
+    private String hostId;
+    private String folderId;
+    private List<String> childrenFolderIds;
+    private List<? extends NavResult> children;
+    private User sysuser = null;
+
+
+    public NavResult(NavResult navResult) {
+        this(navResult.getParentStr(), navResult.getHostId(), navResult.getFolderId(), navResult.getLanguageId());
+    }
+
+    public String getFolderId() {
+        return folderId;
+    }
+
+    public String getParentStr() {
+        return this.parent;
+    }
+
+    public NavResult(String parent, String hostId, String folderId, Long languageId) {
+        this.hostId = hostId;
+        this.folderId = folderId;
+        this.parent = parent;
+        this.languageId = languageId;
+
+
+        title = href = "";
+        order = 0;
+
+        try {
+            sysuser = APILocator.getUserAPI()
+                .getSystemUser();
+        } catch (DotDataException e) {
+            Logger.warn(this, e.getMessage(), e);
+        }
+
+    }
+
+
+
+    public NavResult(String parent, String host, Long languageId) {
+        this(parent, host, null, languageId);
+    }
+
+    public String getTitle() throws Exception {
+        if (title.startsWith("$text.")) {
+            ViewRenderTool render = new ViewRenderTool();
+            render.setVelocityEngine(VelocityUtil.getEngine());
+            return render.eval(title);
+        } else {
+            return title;
+        }
+    }
+
+    public String getHostId() {
+        return hostId;
+    }
+
+    public void setHostId(String hostId) {
+        this.hostId = hostId;
+    }
+
+    public void setTitle(String title) {
+        this.title = title;
+    }
+
+
+    public String getHref() {
+        return href;
+    }
+
+    public boolean isShowOnMenu() {
+        return showOnMenu;
+    }
+
+    public void setShowOnMenu(boolean showOnMenu) {
+        this.showOnMenu = showOnMenu;
+    }
+
+    public void setHref(String href) {
+        this.href = href;
+
+    }
+
+    public boolean isCodeLink() {
+        return this.codeLink != null;
+    }
+
+
+    public String getCodeLink() {
+        return codeLink;
+    }
+
+    public void setCodeLink(String codeLink) {
+        this.codeLink = codeLink;
+    }
+
+
+
+    public int getOrder() {
+        return order;
+    }
+
+    public void setOrder(int order) {
+        this.order = order;
+    }
+
+
+
+
+    public void setChildren(List<NavResult> children) {
+        this.children = children;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Store a copy of "children". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2384)